docs/cloud/pouta/api-access.md

```diff
@@ -0,0 +1,140 @@
+# Pouta access through OpenStack APIs
+
+This article introduces access to Pouta through OpenStack APIs. OpenStack 
+APIs provide access to all OpenStack components and their resources, such 
+as nova (compute), glance (VM images), keystone (authentication), swift 
+(object storage), cinder (block storage), and neutron (networking). 
+
+OpenStack APIs are RESTful and there are multiple ways to interact with 
+them which include using the command-line tools (e.g., `openstack`) or 
+using direct HTTP requests (e.g., with `curl`) or using one of the 
+client libraries (e.g., the `openstacksdk`). 
+
+In this article, we briefly look into using cURL to make direct HTTP requests 
+to the OpenStack APIs  and move to check how to use the `openstacksdk` to automate 
+some mundane tasks as a demonstration. 
+
+### Pouta access through cURL
+
+Before we can use cURL or any other client to make API requests we need 
+to set some environment variables which hold our credentials to Pouta. This 
+can be done by running a script that you can download from the 
+[Pouta web interface](https://pouta.csc.fi/dashboard/project/api_access/) 
+after logging in. See more details from [here](install-client.md).
+
+Once you have the script with your credentials (`<project_name>-openrc.sh`) from 
+the web UI, you can add the environment variables by running the script as:
+
+`source <project_name>-openrc.sh`
+
+And supplying your CSC account username and password when prompted. After this 
+You coud make requests to the Pouta cloud. Normally you would start with 
+authenticating yourself as: 
+
+```
+curl -i   -H "Content-Type: application/json"   -d '
+{"auth": {
+    "identity": {
+      "methods": ["password"],
+      "password": {
+        "user": {
+          "name": "'$OS_USERNAME'",
+          "domain": {"name": "'$OS_USER_DOMAIN_NAME'"},
+          "password": "'$OS_PASSWORD'"
+        }
+      }
+    },
+    "scope": {
+      "project": {
+        "domain": {"id": "'$OS_PROJECT_DOMAIN_ID'"},
+        "name": "'$OS_PROJECT_NAME'"
+      }
+    }
+  }
+}' "$OS_AUTH_URL/auth/tokens?nocatalog" | python -m json.tool
```

Comment:
`| python -m json.tool` must be removed:

```
$ curl -i   -H "Content-Type: application/json"   -d '
{"auth": {
    "identity": {
      "methods": ["password"],
      "password": {
        "user": {
          "name": "'$OS_USERNAME'",
          "domain": {"name": "'$OS_USER_DOMAIN_NAME'"},
          "password": "'$OS_PASSWORD'"
        }
      }
    },
    "scope": {
      "project": {
        "domain": {"id": "'$OS_PROJECT_DOMAIN_ID'"},
        "name": "'$OS_PROJECT_NAME'"
      }
    }
  }
}' "$OS_AUTH_URL/auth/tokens?nocatalog" | python -m json.tool
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1087  100   722  100   365   1885    953 --:--:-- --:--:-- --:--:--  1890
No JSON object could be decoded
```

This is due to the fact that `-i` is used to get the headers. And the headers are not a valid JSON object.

Reply:
Good catch. removed `-i` and added `-v -s`. The piping now works fine.